datashader/pipeline.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, division, print_function
+
+"""
+Declarative interface to Datashader.
+
+Provides a configurable pipeline that makes it simpler to specify
+individual stages independently from the others, Currently does not
+add any new underlying capabilities, with all functionality also
+available using the default imperative interface provided by other
+files.
+"""
+
+import param
+
+from . import transfer_functions as tf
+from . import reductions
+from . import core
+
+
+class Interpolate(param.Parameterized):
+    """
+    Parameterized function object to interpolate colors from a scalar input.
+    """
+
+    low = param.Parameter(default="lightpink", doc="""
+        Color string or tuple specifying the starting point for interpolation.""")
+
+    high = param.Parameter(default="red", doc="""
+        Color string or tuple specifying the ending point for interpolation.""")
+
+    how = param.Parameter(default="log", doc="""
+        Function object or string specifying how to map from a scalar into color space.""")
+
+    def __call__(self, agg):
+        return tf.interpolate(agg, self.low, self.high, self.how)
+
+
+class DatashaderPipeline(param.Parameterized):
+    """
+    Configurable datashading pipeline.  Allows each element of the pipeline
+    to be specified independently without code duplication, e.g. to show the
+    effect of varying that element while keeping the rest of the pipeline
+    constant.
+
+    The pipeline is roughly:
+
+    1. create canvas of the requested size
+    2. aggregate using the specified x and y fields, aggregate field, and agg_fn
+    3. apply the specified transfer_fns, if any, in order
+    4. apply the specified color_fn to translate each resulting aggregate into a color
+    5. return the result as an image
+    """
+
+    data = param.Parameter(doc="""
+        Object supporting columnar-style data access, such as a Pandas
+        dataframe.""")
+
+    x = param.String("x", doc="""
+        Name of the field in the supplied data object to use for the x coordinate.""")
+
+    y = param.String("y", doc="""
+        Name of the field in the supplied data object to use for the y coordinate.""")
+
+    agg = param.String("count", doc="""
+        Name of the field in the supplied data object to use for aggregation.""")
+
+    agg_fn = param.Callable(reductions.count, doc="""
+        Function for aggregating pixel-bin contents into a scalar.""")
```

Comment:
I went ahead and made it work like that, in part because I think the previous interface was halfway in between a Charts-like interface and a datashader-like interface.  I think it makes more sense to have an interface that closely mirrors what datashader supports (which it does now), and another (perhaps more limited) interface that closely mirrors Charts.Scatterplot.  There's very little code involved in either approach, and we can then decide which one is more useful in practice and then keep one or both of them as seems appropriate.